doc/manual/src/store/index.md

```diff
@@ -0,0 +1,84 @@
+# Nix Store
+
+A Nix store is a collection of *store objects* with references between them.
+It supports operations to manipulate that collection.
+
+## Store Object
+
+A store object consists of
+
+  - A [file system object] as data
+  - A set of [store paths](@docroot@/glossary.md#gloss-store-path) as references to other store objects
```

Comment:
Note that in Nix, only top-level FSOs have references. We do not keep track of the references of subdirectories. So references are really metadata that the store keeps per path.

Reply:
Yes this was supposed to be the "fields" of a store object, the first item is the *root* file system object.

(Or equivalently, from a functional programming perspective, "has" vs "refers to" doesn't really mater, and a file system object conceptually includes its children.)

No  - A set of [store paths](@docroot@/glossary.md#gloss-store-path) as references to other store objectst really sure how to phrase this right, but do want to keep it simple, maybe:

```markdown
  - A [file system object] tree
```
on the previous line?

Reply:
(FWIW between hardlinks, FUSE, and other such things, I think "has own copy" vs "refers to with implementation-level reference (as opposed to store path)" are not really interface-level distinctions, since this stuff is immutable.